modules.f90

```diff
@@ -1879,6 +1882,35 @@ function Nu_pidotdot(am,adotoa,Hdot,presnu,presnudot) result (presnudotdot)
         end if
 
     end function Nu_pidotdot
+
+    function Nu_pidotdotdot(am,adotoa,Hdot,Hdotdot,presnu,presnudot,presnudotdot) result (presnudotdotdot)
+        use precision
+        use ModelParams
+
+        real(dl) adotoa,Hdot,Hdotdot,presnu,presnudot,presnudotdot,presnudotdotdot
+        real(dl) d
+        real(dl), intent(in) :: am
+        integer i
+
+        if (am< am_minp) then
+            presnudotdotdot = presnudotdot*( adotoa+Hdot/adotoa )+presnudot*( Hdot+ Hdotdot/adotoa -( Hdot/adotoa )**2 )&
+                &-2._dl/3._dl*const2*( 2._dl*adotoa*Hdot*am +2._dl*adotoa**3*am**2 )
+        else if (am>am_maxp) then
+            presnudotdotdot = 15._dl/2._dl/const*( zeta5/am*( -Hdotdot +3._dl*adotoa*Hdot -adotoa**3 )&
+                & +189._dl/4._dl*zeta7/am**3*( Hdotdot-9._dl*adotoa*Hdot +9._dl*adotoa**3 ) )
+        else
+
+            d=log(am/am_min)/dlnam+1._dl
+            i=int(d)
+            d=d-i
+
+            presnudotdotdot = dddp1(i)+d*(ddddp1(i)+d*(3._dl*(dddp1(i+1)-dddp1(i))-2._dl*ddddp1(i) &
+                -ddddp1(i+1)+d*(ddddp1(i)+ddddp1(i+1)+2._dl*(dddp1(i)-dddp1(i+1)))))
+
+            presnudotdotdot = 0._dl !IW
```

Comment:
not sure what to add here. Where should I look at to implement this cubic spline interpolation?